src/Nethermind/Nethermind.Core/Caching/LruCache.cs

```diff
@@ -83,19 +83,19 @@ public bool TryGet(TKey key, out TValue value)
         }
 
         [MethodImpl(MethodImplOptions.Synchronized)]
-        public void Set(TKey key, TValue val)
+        public bool Set(TKey key, TValue val)
         {
             if (val is null)
             {
-                Delete(key);
-                return;
+                return Delete(key);
             }
 
             if (_cacheMap.TryGetValue(key, out LinkedListNode<LruCacheItem>? node))
             {
                 node.Value.Value = val;
                 _lruList.Remove(node);
                 _lruList.AddLast(node);
+                return false;
```

Comment:
hmmm in Set we have false when we have an element and in Delete, true. Is it intended?